extension/src/aggregate_builder_tests.rs

```diff
@@ -0,0 +1,219 @@
+// Tests for `aggregate_builder::aggregate`. This can't be in the
+// aggregate_builder crate because it requires too much of postgres to actually
+// function
+use aggregate_builder::aggregate;
+
+use pgx::*;
+
+use crate::{
+    palloc::Inner,
+    raw::bytea,
+};
+
+// just about the simplest aggregate `arbitrary()` returns an arbitrary element
+// from the input set. We have three versions
+//  1. `anything()` tests that the minimal functionality works.
+//  2. `cagg_anything()` tests that the config we use for caggs (serialization
+//     but not parallel-safe) outputs the expected config.
+//  3. `parallel_anything()` tests that the parallel version outputs the expected
+//      config.
+#[aggregate] impl toolkit_experimental::anything {
+    type State = String;
+
+    fn transition(
+        state: Option<Inner<State>>,
+        #[sql_type("text")] value: String,
+    ) -> Option<Inner<State>> {
+        match state {
+            Some(value) => Some(value),
+            None => Some(value.into()),
+        }
+    }
+
+    fn finally(state: Option<Inner<State>>) -> Option<String> {
+        state.as_deref().cloned()
+    }
+}
+
+#[aggregate] impl toolkit_experimental::cagg_anything {
+    type State = String;
+
+    fn transition(
+        state: Option<Inner<State>>,
+        #[sql_type("text")] value: String,
+    ) -> Option<Inner<State>> {
+        match state {
+            Some(value) => Some(value),
+            None => Some(value.into()),
+        }
+    }
+
+    fn finally(state: Option<Inner<State>>) -> Option<String> {
+        state.as_deref().cloned()
+    }
+
+    fn serialize(state: Inner<State>) -> bytea {
+        crate::do_serialize!(state)
+    }
+
+    fn deserialize(bytes: bytea) -> Inner<State> {
+        crate::do_deserialize!(bytes, State)
+    }
+}
+
+#[aggregate] impl toolkit_experimental::parallel_anything {
+    type State = String;
+
+    fn transition(
+        state: Option<Inner<State>>,
+        #[sql_type("text")] value: String,
+    ) -> Option<Inner<State>> {
+        match state {
+            Some(value) => Some(value),
+            None => Some(value.into()),
+        }
+    }
+
+    fn finally(state: Option<Inner<State>>) -> Option<String> {
+        state.as_deref().cloned()
+    }
+
+    const PARALLEL_SAFE: bool = true;
+
+    fn serialize(state: Inner<State>) -> bytea {
+        crate::do_serialize!(state)
+    }
+
+    fn deserialize(bytes: bytea) -> Inner<State> {
+        crate::do_deserialize!(bytes, State)
+    }
```

Comment:
Doesn't this need a combine to be truly parallel?  Should marking an aggregate `PARALLEL_SAFE` without providing the required support functions generate a compile time error?

Reply:
🤔 y'know, I've never tried that. I wonder what postgres will do...

Reply:
Apparently that is valid as far as `CREATE AGGREGATE` is concerned, but it won't be parallelizable. Since I currently also don't have checks that an aggregate has either both or neither of `serialize` and `deserialize` I might leave this to a future PR...

Reply:
Well, implementing the check found a bug (I forgot to implement `fn combine()` for the test aggregates) so I think this is worth it.